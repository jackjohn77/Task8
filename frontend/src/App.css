.App {
  text-align: center;
}

.justify {
  text-align: justify;
}

.text-white {
  color: white;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.grey-background {
  background-color: rgb(0, 14, 21);
  color: white;
  padding: 15px;
}

.black-background {
  background-color: rgb(0, 0, 0);
  padding: 15px;
}

.blue-background {
  background-color: rgb(35, 42, 46);
  padding: 15px;
}

.full-height {
  height: 100vh;
}

.active-link {

  color: rgb(251, 251, 251);

  padding: 5px 5px;
  margin: 20px;
  border-bottom: 2px solid #ffffff;

  /* Or any color you prefer */
}

nav a:hover {
  color: #b8b7b7;

}

nav a {
  text-decoration: none;
  color: rgb(255, 255, 255);
  padding: 5px 5px;
  margin-inline: 20px;

}

.active-link:hover {
  color: rgb(251, 251, 251);

}

.product-grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.table-container {
  min-height: 100vh;
  width:100vh;
  
  /* Set to your desired background color */
  padding-bottom: 1rem;
  /* To avoid content being too close to the bottom */
}

