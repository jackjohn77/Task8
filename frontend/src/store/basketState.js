import { createSlice } from "@reduxjs/toolkit";

// Create a slice of the Redux store for basket items
const basketSlice = createSlice({
  name: "Basket", // Name of the slice
  initialState: {
    // Initial state of the slice
    basket: [], // basket items stored here
    shipment: [
      {
        type: "Standard",
        cost: 5.99,
        del: "3-5",
        info: "Standard delivery costs 5.99 and typically takes 3-5 working days. You can track your package and expect delivery from Monday to Saturday between 9 AM and 6 PM.",
        selected: true, // default selected shipment method
      },
      {
        type: "Premium",
        cost: 9.99,
        del: "1-2",
        info: "Premium delivery costs Â£9.99 and ensures your package arrives within 1-2 working days. This service includes tracking and delivers seven days a week, with delivery hours from 8 AM to 8 PM.",
        selected: false,
      },
    ],
    quantity: 0, //Total Quantity of items in the basket
    total: 0, //Total cost of items in the basket
    listId: 0, //ID counter for basket items.
    shipmentCost: 5.99, //Default Shipment cost
  },
  reducers: {
    // Function to add a new item to basket
    basketAdd: (state, action) => {
      const basketItem = state.basket.find(
        (basketItem) =>
          basketItem.title === action.payload.title &&
          basketItem.selectedColour === action.payload.selectedColour
      );
      if (!basketItem) {
        const newId = state.listId + 1;
        const addToState = { ...action.payload, id: newId, quantity: 1 };

        state.basket.push(addToState); //Adds new item to basket
        state.quantity += 1; // Increments basket quantity
        state.total += action.payload.price; // adds the item cost to the current basket cost
        state.listId += 1; // Increments list id.
      } else {
        const basketIndex = state.basket.findIndex(
          (basketItem) =>
            basketItem.title === action.payload.title &&
            basketItem.selectedColour === action.payload.selectedColour
        );
        state.quantity += 1;
        state.total += action.payload.price;
        state.basket[basketIndex].quantity += 1;
      }
    },

    // Function to decrease item quantity in basket
    basketDecrease: (state, action) => {
      if (state.basket[action.payload].quantity > 1) {
        state.total -= state.basket[action.payload].price;
        state.quantity -= 1;
        state.basket[action.payload].quantity -= 1;
      }
    },
    // Function to delete the item from basket in basket
    basketDelete: (state, action) => {
      state.total -=
        state.basket[action.payload].price *
        state.basket[action.payload].quantity;
      state.basket.splice(action.payload, 1);
    },
    // Function to allow the multiple checkbox to only have one item selected
    // and sets the shipment cost to this.
    shippingCheck: (state, action) => {
      state.shipment.forEach((ship) => {
        ship.selected = false;
      });
      state.shipment[action.payload].selected = true;
      state.shipmentCost = state.shipment[action.payload].cost;
    },
  },
});

// Export the action functions to be used in components
export const { basketAdd, basketDecrease, basketDelete, shippingCheck } =
  basketSlice.actions;

// Export the reducer function to be used in the store
export default basketSlice.reducer;
