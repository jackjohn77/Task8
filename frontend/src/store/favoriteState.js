import { createSlice } from "@reduxjs/toolkit";

// Create a slice of the Redux store for favorite items
const favoriteSlice = createSlice({
  name: "Favorite", // Name of the slice
  initialState: {
    // Initial state of the slice
    favorite: [], // favorite items stored here
    quantity: 0, //Total Quantity of items in the favorite
    total: 0, //Total cost of items in the favorite
    listId: 0, //ID counter for favorite items.
  },
  reducers: {
    // Function to add a new item to favorite
    favoriteAdd: (state, action) => {
      const favoriteItem = state.favorite.find(
        (favoriteItem) =>
          favoriteItem.title === action.payload.title &&
          favoriteItem.selectedColour === action.payload.selectedColour
      );
      if (!favoriteItem) {
        const newId = state.listId + 1;
        const addToState = { ...action.payload, id: newId, quantity: 1 };

        state.favorite.push(addToState); //Adds new item to favorite
        state.quantity += 1; // Increments favorite quantity
        state.total += action.payload.price; // adds the item cost to the current favorite cost
        state.listId += 1; // Increments list id.
      } else {
        const favoriteIndex = state.favorite.findIndex(
          (favoriteItem) =>
            favoriteItem.title === action.payload.title &&
            favoriteItem.selectedColour === action.payload.selectedColour
        );
        state.quantity += 1;
        state.total += action.payload.price;
        state.favorite[favoriteIndex].quantity += 1;
      }
    },
    
    // Function to delete the item from favorite in favorite
    favoriteDelete: (state, action) => {
      state.total -=
        state.favorite[action.payload].price *
        state.favorite[action.payload].quantity;
      state.favorite.splice(action.payload, 1);
    }
  },
});

// Export the action functions to be used in components
export const { favoriteDelete, favoriteAdd } =
  favoriteSlice.actions;

// Export the reducer function to be used in the store
export default favoriteSlice.reducer;
