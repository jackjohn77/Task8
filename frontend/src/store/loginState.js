import { createSlice } from "@reduxjs/toolkit";

// Create a slice of the Redux store for login details
const loginSlice = createSlice({
  name: "Login", // Name of the slice
  initialState: {
    // Initial state of the slice
    loggedIn: false,
    username: "",
  },
  reducers: {
    // logs the user in 
    login: (state, action) => {
      state.username = action.payload; // Set the username
      state.loggedIn = true; // Set loggedIn to true
    },
    //logs the user out
    logout: (state) => {
      state.username = ""; // Clear the username
      state.loggedIn = false;
    },
  }
});

// Export the action functions to be used in components
export const { login, logout } = loginSlice.actions;

// Export the reducer function to be used in the store
export default loginSlice.reducer;
