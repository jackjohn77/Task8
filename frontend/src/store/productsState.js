import { createSlice } from "@reduxjs/toolkit";

// Create a slice of the Redux store for product items
const productsSlice = createSlice({
  name: "Products", // Name of the slice
  initialState: {
    // Initial state of the slice
    products: [
      {
        id: 1,
        image: "images/cap1.png",
        title: "Wave Cap",
        description: "Wave Cap in various colours with snap-back",
        price: 9.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 2,
        image: "images/cap2.png",
        title: "Pizza Planet Cap",
        description: "Pizza Planet cap in various colours with snap-back",
        price: 11.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 3,
        image: "images/cap3.png",
        title: "LA Cap",
        description: "LA Cap in various colours with snap-back",
        price: 10.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 4,
        image: "images/cap4.png",
        title: "Focus Cap",
        description: "Focus Cap in various colours with snap-back",
        price: 15.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 5,
        image: "images/cap5.png",
        title: "Sun Flower Cap",
        description: "Wave Cap in various colours with snap-back",
        price: 8.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 6,
        image: "images/cap6.png",
        title: "Croissant Cap",
        description: "Croissant Cap in various colours with snap-back",
        price: 20.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 7,
        image: "images/cap7.png",
        title: "Star Cap",
        description: "Star Cap in various colours with snap-back",
        price: 15.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 8,
        image: "images/cap8.png",
        title: "Humming Bird Cap",
        description: "Humming Bird Cap in various colours with snap-back",
        price: 14.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 9,
        image: "images/cap9.png",
        title: "Bear Golf Cap",
        description: "Bear Golf in various colours with snap-back",
        price: 12.99,
        colours: ["Red", "Blue", "Green"],
      },
      {
        id: 10,
        image: "images/cap10.png",
        title: "Sun Cap",
        description: "Sun Cap in various colours with snap-back",
        price: 11.99,
        colours: ["Red", "Blue", "Green"],
      },
    ],
  },
  reducers: {
  
  },
});


// Export the reducer function to be used in the store
export default productsSlice.reducer;
