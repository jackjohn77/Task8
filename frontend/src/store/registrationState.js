import { createSlice } from "@reduxjs/toolkit";

// Create a slice of the Redux store for registered users
const registrationSlice = createSlice({
  name: "Register", // Name of the slice
  initialState: {
    // Initial state of the slice
    list: [],
  },
  reducers: {
    // Function to add a new user
    add: (state, action) => {
      state.list.push(action.payload); // Add new user to the list
    },
  },
});

// Export the action functions to be used in components
export const { add } = registrationSlice.actions;

// Export the reducer function to be used in the store
export default registrationSlice.reducer;
