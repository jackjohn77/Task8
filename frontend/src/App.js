import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import Menu from "./components/Menu";
import Home from "./pages/Home";
import Products from "./pages/Products";
import About from "./pages/About";
import Login from "./pages/Login";

import Basket from "./pages/Basket";
import "./App.css";

function App() {
  return (
    <div className="grey-background">
      <Router>
        <Menu />
        <Routes>
          {/* Routes to home, product, about page and more*/}
          <Route path="/" element={<Home />} />
          <Route path="/products" element={<Products />} />
          <Route path="/about" element={<About />} />
          <Route path="/login" element={<Login />} />
          
          <Route path="/basket" element={<Basket />} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
