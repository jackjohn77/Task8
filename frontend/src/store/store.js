import { configureStore } from "@reduxjs/toolkit";

// Slice imports
import basketState from "./basketState";
import registrationState from "./registrationState";
import loginState from "./loginState";
import productsState from "./productsState";
import reusableModalState from "./reusableModalState";
const store = configureStore({
  reducer: {
    basket: basketState,
    register: registrationState,
    login: loginState,
    products: productsState,
    reusableModal: reusableModalState,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware(),
  devTools: process.env.NODE_ENV !== "production",
});
export default store;
