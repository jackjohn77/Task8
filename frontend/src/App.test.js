import React from "react";
import App from "./App";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "./store/store";

test("App renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
