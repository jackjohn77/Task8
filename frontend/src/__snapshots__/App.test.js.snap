// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="grey-background"
>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt="Brand logo - Capitals"
          height="100"
          src="Logo_menu.png"
        />
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="ms-auto d-lg-flex text-end navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            href="/products"
            onClick={[Function]}
          >
            Products
          </a>
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            className="nav-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
          <a
            className="nav-link"
            href="/basket"
            onClick={[Function]}
          >
            Basket 
          </a>
        </div>
        <span
          className="text-white ms-auto w-100 justify-content-end text-end navbar-text"
        >
          You are not logged in
        </span>
      </div>
    </div>
  </nav>
  <div>
    <div
      className="App full-height"
    >
      <h2>
        Welcome to the homepage!
      </h2>
      <hr />
      <img
        alt="Store Front"
        className="img-fluid"
        src="store_front.jpg"
        width={800}
      />
    </div>
  </div>
</div>
`;
